build/xtask/src/config.rs

```diff
@@ -287,6 +288,81 @@ impl Config {
             })
             .collect()
     }
+
+    /// Calculates the output region which contains the given address
+    pub fn output_region(&self, vaddr: u64) -> Option<&str> {
+        let vaddr: u32 = match vaddr.try_into() {
+            Ok(v) => v,
+            Err(_) => return None,
+        };
+        self.outputs
+            .iter()
+            .find(|(_name, out)| {
+                vaddr >= out.address && vaddr < out.address + out.size
+            })
+            .map(|(name, _out)| name.as_str())
+    }
+
+    fn mpu_alignment(&self) -> MpuAlignment {
+        // ARMv6-M and ARMv7-M require that memory regions be a power of two.
+        // ARMv8-M does not.
+        match self.target.as_str() {
+            "thumbv8m.main-none-eabihf" => MpuAlignment::Chunk(32),
+            "thumbv7em-none-eabihf" | "thumbv6m-none-eabi" => {
+                MpuAlignment::PowerOfTwo
+            }
+            t => panic!("Unknown mpu requirements for target '{}'", t),
+        }
+    }
+
+    /// Checks whether the given chip's MPU requires power-of-two sized regions
+    pub fn mpu_power_of_two_required(&self) -> bool {
+        matches!(self.mpu_alignment(), MpuAlignment::PowerOfTwo)
+    }
+
+    /// Looks up the `requires` array for the given task or the kernel
+    pub fn requires(&self, name: &str) -> &IndexMap<String, u32> {
+        match name {
+            "kernel" => &self.kernel.requires,
+            name => &self.tasks[name].max_sizes,
+        }
+    }
+
+    /// Suggests an appropriate size for the given task (or "kernel"), given
+    /// its true size.  The size depends on MMU implementation, dispatched
+    /// based on the `target` in the config file.
+    pub fn suggest_memory_region_size(&self, name: &str, size: u64) -> u64 {
+        match name {
+            "kernel" => {
```

Comment:
So, I'm a big fan of allowing `kernel` on the command line, but I wonder if we've gotten a little stringly-typed here. Now everything taking a name needs to handle `kernel`...or not, if it can't, but then it might still get passed `kernel`. This is kind of a validate-over-parsing decision. Would it make more sense to convert the task-or-kernel-name on the command line rather immediately into something like

```rust
enum Name<'a> {
    Kernel,
    Task(&'a str),
 }
 ```
 
 ?

Reply:
I experimented with this a little bit in [a `strongly-typed-task-names` branch](https://github.com/oxidecomputer/hubris/tree/strongly-typed-task-names), but am not totally satisfied with the result – it ripples through a lot of code, and ends up feeling un-ergonomic.  I'd vote to open a ticket for this and revisit later.

Reply:
Fine by me.